### ISSUE TITLE
Incorrect resource name

### ISSUE THREAD

**Issue by aureq** [pulumi/pulumi-google-native#919]
### What happened?

It appears the `iap` service has incorrect resource name, where these resources also contain the API version in their name.

For example `IamBinding` is actually [`V1IamBinding`](https://www.pulumi.com/registry/packages/google-native/api-docs/iap/v1/v1iambinding/) for v1 API and [`V1beta1IamBinding`](https://www.pulumi.com/registry/packages/google-native/api-docs/iap/v1beta1/v1beta1iambinding/) for the v1beta1 API.

A few other resources are affected in the iap service
* IamMember
* IamBinding
* IamPolicy

I didn't find other resources in other services.

### Expected Behavior

The resource names shouldn't contain the API version.

### Steps to reproduce

n/a

### Output of `pulumi about`

n/a

Google Native `0.31.1` is affected.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by danielrbradley** [comment]
It appears this was added approximately a year ago:
- https://github.com/pulumi/pulumi-google-native/pull/653

Only the IAP namespace appears to have this issue.

Unfortunately, fixing this naming would be a breaking change for anyone already using these resources.

We could probably include both the old and fixed named files alongside each other and deprecate the broken ones.

In terms of impact I assume this is mainly a question of consistency rather than it impacting the functionality of the provider?

**Comment by mikhailshilkov** [comment]
@danielrbradley Google Native is still in preview, so a (positive) breaking change isn't a huge issue. We should rename and add an alias to the old names.

**Comment by danielrbradley** [comment]
Yup, ok that makes sense. Will leave on the backlog to consider. Have created a `1.0` label to also keep a track of items we should complete before GA.

**Comment by aureq** [comment]
It may be helpful to use metabase to understand the impact of this change and determine if we could impact customers. A similar check was done for Azure Native 2.0 as I understand it. And if there's little to no customer affected, then could we consider making a release sooner than `1.0`?

**Comment by danielrbradley** [comment]
Yes, it could be made sooner than the 1.0 release, but this feels relatively low impact compared to other bugs which might impact accessing features.

Would be welcome to accept a PR here if you can determine what's causing the odd name.
